### PR TITLE
flag to disable deletion of local parent folder

### PR DESCRIPTION
**Pull request by artsiomtserashkovich**
## Before submitting checklist

- [x] Did you **update the CHANGELOG**? (not for test updates, internal changes/refactors or CI/CD setup)
- [ ] Did you **ask the docs owner** to review all the user-facing changes?

### CODE REVIEW

File: src/neptune/internal/operation_processors/operation_storage.py [hunk 1]
```diff
@@ -55,6 +56,11 @@ def upload_path(self) -> Path:
         return self.data_path / "upload_path"
 
     def cleanup(self) -> None:
+        disable_cleanup = os.getenv(NEPTUNE_DISABLE_LOCALFILES_CLEANUP, "false").lower() in ("true", "1", "t")
+        if not disable_cleanup:
+            self._remove_storage_folder()
+
+    def _remove_storage_folder(self):
         shutil.rmtree(self.data_path, ignore_errors=True)
```
Comment (Raalsky) [inline src/neptune/internal/operation_processors/operation_storage.py:64]: This cannot be impacted by flag, only a few lines below that impact parent directory

File: src/neptune/internal/disk_queue.py [hunk 2]
```diff
@@ -177,7 +178,8 @@ def cleanup_if_empty(self) -> None:
         """
         Remove underlying files if queue is empty
         """
-        if self.is_empty():
+        disable_cleanup = os.getenv(NEPTUNE_DISABLE_LOCALFILES_CLEANUP, "false").lower() in ("true", "1", "t")
+        if self.is_empty() and not disable_cleanup:
             self._remove_data()
```
Comment (Raalsky) [inline src/neptune/internal/disk_queue.py:183]: `shutil.rmtree(path, ignore_errors=True)` needs to stay

File: src/neptune/envs.py [hunk 3]
```diff
@@ -34,6 +34,7 @@
     "NEPTUNE_NON_RAISING_ON_DISK_ISSUE",
     "NEPTUNE_ENABLE_DEFAULT_ASYNC_LAG_CALLBACK",
     "NEPTUNE_ENABLE_DEFAULT_ASYNC_NO_PROGRESS_CALLBACK",
+    "NEPTUNE_DISABLE_LOCALFILES_CLEANUP",
```
Comment (Raalsky) [inline src/neptune/envs.py:37]: That's only parent directory cleanup

File: src/neptune/internal/disk_queue.py (hunk 2 above)
Comment (artsiomtserashkovich) [reply src/neptune/internal/disk_queue.py]: Done

File: src/neptune/envs.py (hunk 3 above)
Comment (artsiomtserashkovich) [reply src/neptune/envs.py]: Done

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,4 +1,9 @@
-## [UNRELEASED] neptune 1.8.3
+## [UNRELEASED] neptune 1.8.2rc0
```
Comment (Raalsky) [inline CHANGELOG.md:1]: 1.8.3 was fine (no need to mention the rcX). Changes after Fixes/Features

File: src/neptune/internal/disk_queue.py [hunk 5]
```diff
@@ -191,7 +192,12 @@ def _remove_data(self):
         except FileNotFoundError:
             files = []
 
-        if len(files) == 0:
+        disable_parent_dir_deletion = os.getenv(NEPTUNE_DISABLE_PARENT_DIR_DELETION, "false").lower() in (
```
[the remark this answers is not in the document]
Comment (artsiomtserashkovich) [reply src/neptune/internal/disk_queue.py]: Done

**Pull request merged**